src/05-derivatives/05-diffusion-derivatives.md

```diff
@@ -63,141 +63,128 @@ field. If a new model is used, common sense should be used to derive a name
-| Shells       | OPTIONAL. Shells that were utilized to fit the model, as a list of b-values. If the key is not present, all shells were used.        |
-| Gradients    | OPTIONAL. Subset of gradients utilized to fit the model, as a list of three-elements lists. If not present, all gradients were used. | ModelDescription | OPTIONAL. Extended information to describe the model. |
-| ModelURL     | OPTIONAL. URL to the implementation of the specific model utilized.                                                                  |
-| Parameters   | OPTIONAL. A dictionary of model parameters (see below for keys specific to each model)                                               |
+| **Key name**     | **Description**                                                                                                                      |
+| ---------------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| Shells           | OPTIONAL. Shells that were utilized to fit the model, as a list of b-values. If the key is not present, all shells were used.        |
+| Gradients        | OPTIONAL. Subset of gradients utilized to fit the model, as a list of three-elements lists. If not present, all gradients were used. |
+| ModelDescription | OPTIONAL. Extended information to describe the model.                                                                                |
+| ModelURL         | OPTIONAL. URL to the implementation of the specific model utilized.                                                                  |
+| Parameters       | OPTIONAL. A dictionary of model parameters (see below).                                                                              |
 
--   Model:
+Parameters that may be stored within the JSON sidecar "Parameters" field depend on the particular model used:
 
-    -   `DTI` :
+-   `DTI` :
 
-        -   `FitMethod` : {`WLS`,`OLS`,`NLLS`,`RESTORE`}
-        -   `RESTORE` : sigma
+    -   `FitMethod` : {`WLS`,`OLS`,`NLLS`,`RESTORE`}
+    -   `RESTORE` : sigma
 
-    -   `DKI` and `WMTI`:
+-   `DKI` and `WMTI`:
 
-        -   `FitMethod` : {`WLS`,`OLS`,`NLLS`}
+    -   `FitMethod` : {`WLS`,`OLS`,`NLLS`}
 
-    -   `CSD`:
+-   `CSD`:
 
-        -   `SphericalHarmonicOrder` : value
-        -   `ResponseFunctionOrder` : value
-        -   `ResponseFunction` : \[1:response_function_order_value,b0mean\]
-        -   `Basis` : {`MRtrix 0.2`,`MRtrix3`,`DESCOTEAUX`}
+    -   `Tissue` : string
+    -   `SphericalHarmonicDegree` : value
+    -   `ResponseFunctionDegree` : value
+    -   `ResponseFunction` : values
+        (1 row per unique *b*-value, with lower *b*-values listed first;
+        1 column per even harmonic degree starting from zero)
+    -   `SphericalHarmonicBasis` : {`MRtrix 0.2`,`MRtrix3`,`DESCOTEAUX`}

```

Comment:
I am the fence here if the articles are specified as proposed by @jchoude users will need to go into the papers and translate the text there into code.  Instead, by pointing to the toolbox more technical information can be found.

Reply:
"Normalized" would be too ambiguous; it's specifically an *orthonormal* SH basis.

For someone without access to these existing software tools, reading those papers would not provide enough detail for that person to be able to properly interpret data encoded in one of these bases. So both that proposal, and the current text, have their own issues.

If we really want a general solution, that can both support data from existing tools and not be tied specifically to software tools, the bases themselves would need to be decomposed:

- Is it orthonormal, or is there a sqrt(2) factor in the m!=0 terms?
   (Or indeed, should we simply throw this out altogether? We're removing support for this from *MRtrix3*, and I doubt any substantial number of people would try to use MRtrix 0.2 with BIDS)

- What is the scaling between DWI amplitude data and the SH representation?

- Is there a -1 scaling between consecutive (even) spherical harmonic degrees?

Reply:
You're right about the fact that people should be using Mrtrix3 with Bids. Even for legacy data that might have been previously processed with Mrtrix 0.2, I think people would be better off re-processing it. In this case, I agree we should drop Mrtrix 0.2, and just use the names you suggested.

Reply:
I am all for reducing the number of dependencies.

Reply:
I'm happy to remove the MRtrix 0.2 basis support also.

The question of naming of the remaining two options nevertheless remains. Some options:

1. `SphericalHarmonicBasis` : [`MRtrix3`, `DESCOTEAUX`]
    (or alternatives; e.g. should both be named after authors, or should both be named after softwares?)

2.  `SphericalHarmonicScaleFactor` : value
      (Some parameter relating to the magnitude scaling between the SH and amplitude basis; for instance, it could define the value of the l=0 coefficient corresponding to the outcome of deconvolving the response function with itself)
      `SphericalHarmonicInvertEveryOtherDegree` : boolean

We're also assuming throughout here both that the order in which the different SH coefficients are provided in a 4D ODF image are the same between these two existing bases, and that this ordering is sufficiently unambiguous to not need to be specified explicitly.

Reply:
@jchoude is this assumption Ok for dipy? 

_"We're also assuming throughout here both that the order in which the different SH coefficients are provided in a 4D ODF image are the same between these two existing bases, and that this ordering is sufficiently unambiguous to not need to be specified explicitly."_

Reply:
So we seem to agree on dropping mrtrix 0.2 @Lestropie do you guys have a way to read/write mrtrix2 tractography into mrtrix3 tractography? If not how difficult would it be for folks to reprocess the data as @jchoude suggests?

Reply:
> @Lestropie do you guys have a way to read/write mrtrix2 tractography into mrtrix3 tractography?

For *ODFs* we do: command `shbasis` makes a decision about whether or not input SH data are or are not in the newer orthonormal basis, and can convert data between the two. We're actually considering removing this command from the repository, since we highly doubt anybody actually requires it any more, but it would still be made available e.g. as a Gist.

For *tractography* data, the `.tck` format remains unchanged since 0.2.